contracts/utils/Cruzo1155Factory.sol

```diff
@@ -0,0 +1,73 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+contract Cruzo1155Factory is Context, Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    event newTokenCreated(
+        address indexed tokenAddress,
+        uint256 indexed newTokenId
+    );
+
+    mapping(uint256 => address) private tokens;
```

Comment:
IMO, this is unnecessary

Reply:
Have some troubles with testing now, i am used to get addr of token from this mapping, so, for short time i added variable called last, to track address to test.
In next commit i will rewrite this and events would be used